packages/blade/src/components/Chip/__tests__/ChipGroup.native.test.tsx

```diff
@@ -0,0 +1,234 @@
+          const { toJSON, getByText } = renderWithTheme(
+            <ChipGroup selectionType={selectionType} intent={intent}>
+              <Chip value="apple">Apple</Chip>
+              <Chip value="mango">Mango</Chip>
+              <Chip value="orange">Orange</Chip>
+            </ChipGroup>,
+          );
+
+          expect(getByText('Apple')).toBeDefined();
+          expect(getByText('Mango')).toBeDefined();
+          expect(getByText('Orange')).toBeDefined();
+          expect(toJSON()).toMatchSnapshot();
+        });
+      });
+
+      it('should propagate the "isDisabled" prop to all children chips', () => {
+        const { getAllByRole } = renderWithTheme(
+          <ChipGroup isDisabled selectionType={selectionType}>
+            <Chip value="apple">Apple</Chip>
+            <Chip value="mango">Mango</Chip>
+            <Chip value="orange">Orange</Chip>
+          </ChipGroup>,
+        );
+
+        const chipRole = selectionType === 'single' ? 'radio' : 'checkbox';
+        getAllByRole(chipRole).forEach((chip) => {
+          expect(chip.props.accessibilityState.disabled).toBe(true);
+        });
+      });
+
+      it('should propagate name prop to children chips', () => {
+        const { getAllByRole } = renderWithTheme(
+          <ChipGroup name="fruits" selectionType={selectionType}>
+            <Chip value="apple">Apple</Chip>
+            <Chip value="mango">Mango</Chip>
+            <Chip value="orange">Orange</Chip>
+          </ChipGroup>,
+        );
+
+        const chipRole = selectionType === 'single' ? 'radio' : 'checkbox';
+        getAllByRole(chipRole).forEach((chip) => {
+          expect(chip.props.name).toBe('fruits');
+        });
+      });
+
+      it('should work in uncontrolled mode', () => {
+        const isSingleSelection = selectionType === 'single';
+        const onChange = jest.fn();
+        const { getByA11yState, getAllByRole } = renderWithTheme(
+          <ChipGroup
+            selectionType={selectionType}
+            name="fruits"
+            defaultValue={isSingleSelection ? 'apple' : ['apple']}
+            onChange={onChange}
+          >
+            <Chip value="apple">Apple</Chip>
+            <Chip value="mango">Mango</Chip>
+            <Chip value="orange">Orange</Chip>
+          </ChipGroup>,
+        );
+
+        const chip = getByA11yState({ checked: true });
+        expect(chip.props.value).toBe('apple');
+
+        const chips = getAllByRole(isSingleSelection ? 'radio' : 'checkbox');
+        const apple = chips.find((chip) => chip.props.value === 'apple');
+        const mango = chips.find((chip) => chip.props.value === 'mango');
+        const orange = chips.find((chip) => chip.props.value === 'orange');
+
+        expect(apple?.props.accessibilityState.checked).toBeTruthy();
+        expect(onChange).not.toBeCalled();
+
+        if (isSingleSelection) {
+          fireEvent.press(mango as ReactTestInstance);
```

Comment:
Let's try to keep the tests as pure as possible. 

Code duplication should be fine, let's keep separate test suites for `single select` and `multi select` rather than adding if statements and dynamically generating the test. 

```
it('should work with controlled state when selectionType=single', () => {
  // code for single selection without any external logic
})

it('should work with controlled state when selectionType=multiple', () => {
  // code for multi selection without any external logic
})
```

Reply:
Updated.